pkg/golinters/gocritic.go

```diff
@@ -254,377 +261,344 @@ func (w *goCriticWrapper) normalizeCheckerParamsValue(p any) any {
 		return rv.Bool()
 	case reflect.String:
 		// Perform variable substitution.
-		return strings.ReplaceAll(rv.String(), "${configDir}", w.cfg.GetConfigDir())
+		return strings.ReplaceAll(rv.String(), "${configDir}", w.lintConfigDirGetter())
 	default:
 		return p
 	}
 }
 
-// TODO(ldez): rewrite and simplify goCriticSettingsWrapper.
-
 type goCriticSettingsWrapper struct {
 	*config.GoCriticSettings
 
 	logger logutils.Log
 
-	allCheckers   []*gocriticlinter.CheckerInfo
-	allCheckerMap map[string]*gocriticlinter.CheckerInfo
+	allCheckers []*gocriticlinter.CheckerInfo
 
-	inferredEnabledChecks map[string]bool
+	allChecks           map[string]struct{}
+	allChecksLowerCased map[string]struct{}
```

Comment:
`*LowerCased` used for `SettingsPerCheck` validation only

Reply:
this should be added in a comment inside the code.

Reply:
done